### ISSUE TITLE
Call of ideas!

### ISSUE THREAD

**Issue by thigazi** [Pyeg/Ideas#5]
People!
I just fired up short time in bitbucket the pyeg-cms project. 

The site is tomorrow online..... we need the call of ideas now.

**Comment by mtayseer** [comment]
Why a new Python CMS?

**Comment by thigazi** [comment]
Not a CMS, it's our site!

**Comment by mahmoudhossam** [comment]
Why is it on bitbucket and not here?

**Comment by thigazi** [comment]
Because it is a website, and not software.
I even don't understand why "github" is used for sharing ideas. totally strange... 

However, the mailman engine is soon setup for all of us, that we can use the mailman server to talk with each others...

Github is used usually for software development, and to speak about errors / issues in the software among with each others. So, usually this is the wrong place to share ideas.

I think we should continue our discussions here:

https://groups.google.com/forum/#!forum/python-eg

Tamer

**Comment by mahmoudhossam** [comment]
Github works okay for me, we're already talking.

Is there anything you can do in a mailing list that you can't do here?

I know it's not usual, but I don't think it's essentially bad.

As for the Google group, it looks dead, no posts this year.

**Comment by thigazi** [comment]
Mahmoud, thanks for your reply!
Yes, it is "reason" oriented. 

All python groups world wide run their own mailinglists on their own sites, which has a sense. 

I am a very "sense" oriented person, the "pyeg.org" domain is up, soon you will find a complete basic site and the registration info for "lists.pyeg.org". 

GitHub is a community site for open source dvelopment.
The reason is very simple. 

Getting organized the way the others are as well. Not more and not less, and you are very welcome to participate in our project with work and effort if you like.

The google group would be the best place to talk, because it's a group, and not a github site. This is what I hate... here.

**Comment by mahmoudhossam** [comment]
I haven't used a mailing list before, but if everyone's on board with this, I don't suppose I have a choice.

@gr33ndata @mtayseer your thoughts?

**Comment by thigazi** [comment]
which will change over the time....

**Comment by gr33ndata** [comment]
I like how things are organized here, as tickets can have priorities and sorted based on number of replies or so. It is not new to use github for hosting similar discussions or even to host static web pages and documentations. See, the Open Knowledge Foundation do the same thing here, https://github.com/okfn/ideas

We still can have a mailing list anyway if you want to, but I believe in using whatever we have already that works without bothering ourselves with setting sophisticated softwares and content management systems.

**Comment by mtayseer** [comment]
**tl;dr** I'm OK with both options

**Long version** Before I post the first issue here, I thought about the best way to discuss ideas. There are many ways: mailing lists, forums, issues on GitHub , Facebook groups, subreddits, etc. There is no one which makes more "sense" than the other. I saw different communities use these tools successfully, because it's not about the tool itself. We are already using a GitHub issue for this discussion!

I've an interesting story to tell: in the college we wanted to start a community for learning. People wanted to create a logo for the group. Tens of people participated with design & hundreds of emails discussing the tiniest detail about each design. Finally we selected a logo, then guess what? Less than 20 people kept contributing.

My message is: there are things which are more important to the community than the tools

**Comment by thigazi** [comment]
Mohammed,
I advise to continue all discussions on:

https://groups.google.com/forum/#!forum/python-eg

I tell you why. I am working a lott with my Mailclient. And to login everytime at github to answer you is for me too stressfull. 

At Google Groups, or even if you run your own mailman engine, you can answer from your MTA client, and it is posted and recorded on the web. 

If something makes my life easier, then I of course prefer it instead of logging myself in a stupid website to post something.

**Comment by thigazi** [comment]
Mohammed,
Ask Tarek to join "python-eg", you are welcome to participate here until everything's up. 

You ask "why" tools?!

Professional people, orgaize and collaborate professionally.....
Time is very useful, to loose it in a nonsense way is a awful way.....

**Comment by mahmoudhossam** [comment]
@thigazi The way I see it, you're the only one who has a problem with using Github for discussion.

As for email, someone already [used email](https://github.com/Pyeg/Ideas/issues/4#issuecomment-31086736) to reply to an issue on this repo, so you can use that as well, no login required.

Maybe we can set up a mailinglist and a website after we've created some content (projects, documentation...etc.) and not before?

**Comment by aessam** [comment]
Funny thread.

Communication is efficient by the people involved, tools and procedures are
for optimization.

So we wasted enough time talking about the tools.

Painful when we don't practice what we preach.

Part of this thing is the spirit and be a good team player wherever we are
playing.

**Comment by thigazi** [comment]
You are making yourself funny about others... what about you?!
Participate. In the previous posts I mentioned that we need layout. 

Perhaps you can make the layout and the CSS stuff and prepare it for me. 
As well I setup mailman and coding for the frontend. 

Sorry, but not any here has done a bit.... don't you think so ?!
Show that you are doing something instead of making yourself funny about the thread.

**Comment by mtayseer** [comment]
> ... I am a very "sense" oriented person ...
> ... stupid website to post something ....
> ... Time is very useful, to loose it in a nonsense way is a awful way ........
> ... Sorry, but not any here has done a bit.... don't you think so?! ...
> ... Show that you are doing something instead of making yourself funny about the thread ....

Tamer,
I'm tired of you lecturing us arrogantly about sense, intelligence & value of time. I found [django-cms themes](http://www.djangocmsthemes.com/) & I was going to send them to you & offer help with django-cms, to find you say that you're the only one doing something here?! If you read the discussions we had earlier on this _stupid website_ you will find that we had some ideas & I already posted a project I made before to use as a base. 

You should show respect for the time other people spent here!

**Comment by gr33ndata** [comment]
![facebook_like_thumb](https://f.cloud.github.com/assets/135069/1825373/e73dcc8e-71b5-11e3-995d-8b98dcbc2cee.png)

Let's end this discussion here, or let me use the github lingo and say, let's close this issue now

**Comment by thigazi** [comment]
Mohamed,
1. I am not making myself funny of anybody at all. I just want to get with this website forward, as I believe we need to get finished as every other python community has. 
2. Since 8 days we are speaking about the site, and we are not 1 step getting forward. You came along with other concepts, but we are speaking the whole time about python. 

Why should I want to make a site for a bunch of programming languages with which we, and the others are not in love with, or out of my concept ?! 

This is like, a football trainer needs to start training a volleyball team.... I thought you guys are python coders, no ?! If somebody is a C++ expert, he can make a community which he moderates and understand best with. 

Instead of coming up with a tons of ideas all the time, why not working out the 1st parts and go from there step by step ???

You gave me now the link with the django template, but which one do we like to choose for the site ?!
By the way, there is no real "django template", there is only a CSS styled template, and that's all. 

It is spitted by 2: 

layout.html
LeftNav.html
Content.html

that's it. 

But when you don't tell me what you want me todo, I am looking like a clown and don't come forward. 
It is only a site, nut a real big application. 

I make you an offer to split the work:
1. look please for a template of, all of you guys want. I cannot decide for you. YOU have to like it, not me. 
2. the subjects are important. 

I advise: 

Welcome
Who we are
Learning Python
Mailinglist
disclaimer

(please vote)

otherwise tell me what you want, and I take care of it, and please define the content for me. 

This is what we call "teamwork". 

But if I say something you don't like to hear you start to get upset ?!
I am willing to help and push this to success that we have A REAL PYTHON COMMUNITY with continues visitors, and not this facebook stuff. 

Please stay focused to the subject, that we get this site finished, and then we can speak about further modules, and features we could continuously implement. 

This can be only accomplished, when all of us participate. 

I'm offering you my help in coding and sharing my KNOW-HOW with you. 
But there must a little bit come back from you. 

Not more and not less I am asking. 

2nd, we have the beautifull python-egypt group, which Tarek activated for me today (thanks again). 
IT IS, and believe me, much easier to answer from a MTA client directly instead of logging somewhere in everytime. 

You spoke about respect.....

respect comes with actions, not those who are making themselfs funny about a discussion without participating. 

I think we should stop criticizing each others, please let us focuse for our targets. 
It is my small egyptian dream, to have in Egypt a real living Python Community alive. 
It would be sad, if we split up now in anger, don't you think the same as I do ?!

**Comment by thigazi** [comment]
For any misunderstandings, of course I excuse. 
Writing through Mail or Web causes always misunderstandings, because there is no continues dialogues.....

**Comment by mtayseer** [comment]
For the theme, I prefer the [simple theme](http://www.djangocmsthemes.com/themes/simple/)

Please share the pyeg-cms repo with us. We can create issues & assign them there. I think we should move it to GitHub to keep all our projects in one place.

**Comment by thigazi** [comment]
What is your bitbucket username ?! 
I will add you immediatly. 

Also I changed the accounts to full write access in the repisotory, which I haven't seen before.

**Comment by thigazi** [comment]
I found you, and added you to the pyeg group on bitbucket.

**Comment by mtayseer** [comment]
Great.

The issues tab is not visible when I view the repo. How can I create issues?

On Tue, Dec 31, 2013 at 2:50 PM, thigazi notifications@github.com wrote:

> I found you, and added you to the pyeg group on bitbucket.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/Pyeg/Ideas/issues/5#issuecomment-31393748
> .

## 

_Mohammad Tayseer_
My site: http://mtayseer.net
My projects: http://mtayseer.net/projects/, https://github.com/mtayseer
Twitter: http://twitter.com/m_tayseer

**Comment by thigazi** [comment]
We can't. I think i will push the code up here!

the bitbucket account doesn't give us that much freedom.

Tarek! Please open a repository "pyegcms", I will fire up the code in the branch "develop". 

After I fired up the code here, I will close the repisotory on bitbucket. 

Tamer

**Comment by gr33ndata** [comment]
Done, you should have access to it now
https://github.com/Pyeg/pyegcms/

**Comment by thigazi** [comment]
ERROR: Permission to Pyeg/pyegcms.git denied to thigazi.

You have to give me write access, my ssh key is in my accounts profile.

**Comment by thigazi** [comment]
however, I need beside pull also push access. 

Tamer

**Comment by ghost** [comment]
However, after you gave me right access I will fire up the code.

Tamer

**Comment by gr33ndata** [comment]
try again now

**Comment by ghost** [comment]
I did, and should be available now!

Tamer

**Comment by sheshtawy** [comment]
I cannot join google group .. don't have permission .. may I have it plz ?

**Comment by thigazi** [comment]
Ahlan Hisham!

Ask Tarek,
He will activate your account.

Tamer

**Comment by mlatief** [comment]
@Sheshtawy welcome on board :)

**Comment by sheshtawy** [comment]
@mlatief thanks :D 

@gr33ndata add me to google group please :D

**Comment by thigazi** [comment]
Hisham!

Tell us a bit about you. Are you a python coder?!

Tamer

**Comment by gr33ndata** [comment]
would you please send me your email to add you

**Comment by sheshtawy** [comment]
@thigazi  .. I think I am .. 
check this to know more about me :) 
http://about.me/hisham.elsheshtawy

**Comment by thigazi** [comment]
Beautiful!

Welcome !!!

Tamer

**Comment by gr33ndata** [comment]
The first time I met Hisham was in a competition run by Google Developers Group in Cairo, and he and his team won the first place after building a web application using Python and Google App Engine.

**Comment by thigazi** [comment]
Cool!
I am happy to hive him in the club!

Welcome HISHAM! I am happy to know you!

Tamer

**Comment by sheshtawy** [comment]
Thanks Guys :smile:  .. it's really pleasure to join you all ... and I am so happy to be part of such a great interesting community

**Comment by thigazi** [comment]
Hi Hisham!

I am online working on a customer project. I have to see to get the
template stuff finished as well, getting the tinymce editor to work as
well to get the site finished.

If you'd love to join, we welcome you!

Do you work with git on shell as I do?! If not, here is a small guide

very important (that you fetch the develop branch from the beginning):

git clone -b develop git@github.com:Pyeg/pyegcms.git

don't forget to set your username and email address inside the cloned
folder with:

git config user.name "Your Full Name"
git config user.email "my@mailaddress.com"

then git pull (to get the changes)

and

git push, should do it!

Ask Tarek to figure out from the beginning if he had given you write
access or not.

Tamer

**Comment by sheshtawy** [comment]
@thigazi  could you please give me more details and explain your first paragraph :D ? 

thanks ..

**Comment by thigazi** [comment]
I guess you are working on a linux machine.

If you have never worked before with "git", you must 1st generate a ssh
key. this key you put in your profile.

1st - making your SSH-Key and put the public key in your git account:

ssh-keygen -t rsa -b 4096

which will store your keys in:

/home/yourusername/.ssh/

my folder looks like this:

/home/tamer/.ssh

there you have 3 files:

tamer@tux ~/.ssh $ ls -lA
total 12
-rw------- 1 tamer tamer 3326 Sep 21 10:27 id_rsa
-rw-r--r-- 1 tamer tamer  735 Sep 21 10:27 id_rsa.pub
-rw-r--r-- 1 tamer tamer 2798 Dec 31 15:09 known_hosts

the content from the public key, you put in your github profile, under
ssh keys (i believe)

2nd - clone the repisetory

there are 2 branches, develop and master
we are working on the developer branch.

you just clone the remote repository on your machine. But to get the
"develop" branch and NOT the master you add the -b flag with the name of
the repository you want to work with.

You do that with:

git clone -b develop git@github.com:Pyeg/pyegcms.git

3rd - setting username and mailaddress

You need to set inside the cloned directory the username and emailadress

go inside the cloned directory and type:

git config user.name "Hisham Elsheshtawy"
git config user.email "your@mail.com"

replace your@mail with the mailaddress at which you left at github.

4th - adding, deleting and making the changes in the repository

git add \* => adds all changed files
git rm filename => deletes file and marks it for deletion

git commit => opens the editor
git commit -m "Message" => does it directly.

then

git push for pushing the changes on github.

5th - more about github online

I advise you heavily reading the online github official documentation.

http://git-scm.com/documentation

I'm sure you'll be getting very familiar in a short time.

Tamer

**Comment by mos3abof** [comment]
Dude, the guy is on Github, he definitely KNOWS git :D

He was asking about the website details, not git details.

**Comment by sheshtawy** [comment]
@mos3abof  .. whatever .. it's an awesome demo.
@thigazi  .. thanks for attitude .. you cannot imagine how much opportunities wasted by people who didn't help beginners or people new to context like that. 
and I need details about site beside these instructions.
and btw I am not so used to git yet .. but I am adventurer :monkey:  :smile:

**Comment by thigazi** [comment]
I saw it only from my side as well.

Longer time ago I worked for an advertising agency, and I wasn't very
much involved with "git" because I wasn't working in the past in a team.

I just coded alone for individual customers, and nobody showed me how to
use git, until I red the entire documentation.

Back to the subject:

We are using "django-cms" for the website. The sample is online at
"pyeg.org" which I run through mod_wsgi on latest apache version on my
server.

You can login at "pyeg.org/admin"

Send the "python-egypt" group a mail, I'll respond you, and give you the
password.

We got currently have 1 small problem, the tinymce editor, which I still
have to take care off.

Perhaps you can figure out what I did wrong, I have to ask the
"django-cms" group. I am not a django expert, so this is for me new land
as well.

I choosed "django-cms" because it looks simple.

I advise you setting up a "virtualenv" and install through pip
"django-cms".

If you have some free time and want to participate and need help setting
up the environment, I am happy to help you.

Send the "python-egypt" group a mail, I'll respond and we go through
this step by step that you have the environment setup as quickly as
possible.

I am offering it to you, because I don't know how you work, and perhaps
I can make your life a bit easier.

Tamer

**Comment by mtayseer** [comment]
@thigazi This is what I did to make it work
1. Changed `SITE_ID` to be `1`
2. Removed `djangocms_text_tinymce` from `INSTALLED_APPS`
3. Added `cms.plugins.text` to `INSTALLED_APPS`

I also added this snippet to the end of `settings.py`

``` python
try:
    from local_settings import *
except ImportError:
    pass
```

which allows you to create `local_settings.py` to include your own customizations, e.g. in my case I'm using sqlite. This allows you to fix issues like [this](https://github.com/Pyeg/pyegcms/issues/2)

My changes are [in this commit](https://github.com/Pyeg/pyegcms/commit/1de3a0499f0253e921be2768059628efde9eb91e)

PS: I prefer to move the conversation about PyEgCMS to issues in that repo

**Comment by thigazi** [comment]
Cool it works!

I will continue this night with the templating stuff and see that I have
the mailman engine to be hocked into the site then.

...as usual, I will commit the changes when I'm done before firing it up.

Thank you Mohammad!

To Tarek:
Let us delete the "ideas" repository and start commenting inside the
pyegcms repository

Tamer

**Comment by gr33ndata** [comment]
Ideas is for suggesting project to work on or ways to develop the community. 
Let the pyegcms repository be for dicussing the CMS development and deplyment itself

**Comment by thigazi** [comment]
Okay!
